cajita/unit_test/tstGlobalGrid.hpp

```diff
@@ -390,6 +391,196 @@ void gridTest2d( const std::array<bool, 2>& is_dim_periodic )
+    }
+
+    // Check the number of edges entires
+    {
+        // Number of I edges
+        EXPECT_EQ( global_grid->globalNumEntity( Edge<Dim::I>(), Dim::I ),
+                   global_num_cell[Dim::I] );
+
+        EXPECT_EQ( global_grid->globalNumEntity( Edge<Dim::I>(), Dim::J ),
+                   global_num_cell[Dim::J] + 1 );
+
+        EXPECT_EQ( global_grid->globalNumEntity( Edge<Dim::I>(), Dim::K ),
+                   global_num_cell[Dim::K] + 1 );
+
+        // Number of J edges
+
+        EXPECT_EQ( global_grid->globalNumEntity( Edge<Dim::J>(), Dim::I ),
+                   global_num_cell[Dim::I] + 1 );
+        EXPECT_EQ( global_grid->globalNumEntity( Edge<Dim::J>(), Dim::J ),
+                   global_num_cell[Dim::J] );
+
+        EXPECT_EQ( global_grid->globalNumEntity( Edge<Dim::J>(), Dim::K ),
+                   global_num_cell[Dim::K] + 1 );
+
+        // Number of K edges
+
+        EXPECT_EQ( global_grid->globalNumEntity( Edge<Dim::K>(), Dim::I ),
+                   global_num_cell[Dim::I] + 1 );
+
+        EXPECT_EQ( global_grid->globalNumEntity( Edge<Dim::K>(), Dim::J ),
+                   global_num_cell[Dim::J] + 1 );
+        EXPECT_EQ( global_grid->globalNumEntity( Edge<Dim::K>(), Dim::K ),
+                   global_num_cell[Dim::K] );
+    }
+
+    // Check the partitioning. The grid communicator has a Cartesian topology.
+    int comm_size;
+    MPI_Comm_size( MPI_COMM_WORLD, &comm_size );
+    auto grid_comm = global_grid->comm();
+    int grid_comm_size;
+    MPI_Comm_size( grid_comm, &grid_comm_size );
+    int grid_comm_rank;
+    MPI_Comm_rank( grid_comm, &grid_comm_rank );
+    EXPECT_EQ( grid_comm_size, comm_size );
+    EXPECT_EQ( global_grid->totalNumBlock(), grid_comm_size );
+    EXPECT_EQ( global_grid->blockId(), grid_comm_rank );
+
+    std::vector<int> cart_dims( 3 );
+    std::vector<int> cart_period( 3 );
+    std::vector<int> cart_rank( 3 );
+    MPI_Cart_get( grid_comm, 3, cart_dims.data(), cart_period.data(),
+                  cart_rank.data() );
+    for ( int d = 0; d < 3; ++d )
+    {
+        EXPECT_EQ( cart_period[d], is_dim_periodic[d] );
+        EXPECT_EQ( global_grid->dimBlockId( d ), cart_rank[d] );
+        EXPECT_EQ( global_grid->dimNumBlock( d ), ranks_per_dim[d] );
+    }
+
+    auto owned_cells_partitioner =
+        partitioner.ownedCellsPerDimension( global_grid->comm() );
+    for ( int d = 0; d < 3; ++d )
+        EXPECT_EQ( global_grid->ownedNumCell( d ), owned_cells_partitioner[d] );
+
+    // Update partitioner and check num cell and global offset
+    auto part = partitioner.getCurrentPartition();
+    for ( int d = 0; d < 3; ++d )
+        for ( int id = 1; id < ranks_per_dim[d]; id++ )
+            part[d][id] += 1;
+
+    partitioner.initializeRecPartition( part[0], part[1], part[2] );
+
+    std::array<int, 3> new_owned_num_cell;
+    std::array<int, 3> new_global_cell_offset;
+    partitioner.getOwnedCellInfo( global_grid->comm(), global_num_cell,
```

Comment:
I don't feel it quite necessary here for the dense grids for several reasons.
1) This function is already called in the constructor of GlobalGrid (in line 54 in `Cajita_ClobalGrid_impl.hpp`) and serves as the base of the rest of the code and checks. So basically other tests have already covered this function.
2) This unit test is for GlobalGrid but not for Partitioner, and the purpose of lines 554 - 581 is to check the behavior of function `setNumCellAndOffset` in `GlobalGrid`. 
Here's the reason why this function is called for the sparse grid but not for the dense grid here: 
Since we have a dynamic load balancer for the sparse grid, the sparse partition can tell the owned cell and the offset information to the global grid, providing `setNumCellAndOffset` a more reasonable source. But the dense grid partitioner will only provide the `ranks_per_dim` info and assume an average partition. So calling this function will only generate the same `ownedNumCell` and `globalCellOffset` as before. This doesn't fit the purpose of testing `setNumCellAndOffset` with new settings.